### PR TITLE
Closes #109 platform admin

### LINKED ISSUE #109: Platform Admin Views Category Show
As a platform admin,
When I visit "/categories/#{category.name}",
Then I see a listing of all "bidding open" jobs in that category,
And each job has:

A link to its show page that is job name
A description
Poster's name as link to profile
Poster's image
Number of bids on the job
Price range of bids
The ending date of the job auction
Location of the job
The estimated duration tag
Links to other categories on side bar

### LINKED ISSUE #110: Platform Admin Views Job Show - Bidding Open
As a platform admin,
When I visit "/users/#{lister_user_name}/jobs/#{job.id}",
And the bidding for that job is open,
I see the "Place a Bid" accordion,
And I do not see the "Log in to Place a Bid" button in its place,
And I do see "Current Bids" table, containing columns:

Bidder Name
Price
Duration Estimate
Comment
Action

And I do see info for the job:
A link back to the category show
The name of the job
The full description of the job
Estimated duration tag
"Must be completed by" date
"Bidding ends" date
Total # of Bids
Bid Range
Location of the job

And I do see the Lister's Information showing:
Poster's name as link to profile
Poster's image
Rating
Number of active jobs
Number of completed jobs
"Active Since" date

And I do see a "Delete" button

### LINKED ISSUE #116: Contractor Dashboard Bug - "No Listings"
When I log in as a contractor and go to my dashboard, I see all of my bids in the table but I also see, "You have not listed any jobs yet." at the bottom of the table

### PR DESCRIPTION
**Pull request by ToniRib**
Platform Admin can view the category show page (no real change here) and they can also view the job show page. The Platform Admin gets a "Delete Job" button on every job show page, which will delete the job and all associated bids.

Also did some refactoring of the current bids table so that more logic is moved into the JobPresenter.

closes #110 
closes #109 
closes #116

### CODE REVIEW

File: test/integration/platform_admin_views_job_show_page_test.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  test "platform admin views job show page when bidding is closed" do
+    bid = create(:bid)
+    bidder = bid.user
+    job = bid.job
+    job.bidding_closed!
+    lister = job.lister
+    platform_admin = create(:platform_admin)
+
+    ApplicationController.any_instance.stubs(:current_user).returns(platform_admin)
```
Comment (ToniRib) [inline test/integration/platform_admin_views_job_show_page_test.rb:42]: Not much I can do about this one.

File: test/integration/platform_admin_views_job_show_page_test.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+class PlatformAdminViewsJobShowPageTest < ActionDispatch::IntegrationTest
+  test "platform admin views job show page while bidding is open" do
+    bid = create(:bid)
+    bidder = bid.user
+    job = bid.job
+    job.bidding_open!
+    lister = job.lister
+    platform_admin = create(:platform_admin)
+
+    ApplicationController.any_instance.stubs(:current_user).returns(platform_admin)
```
Comment (ToniRib) [inline test/integration/platform_admin_views_job_show_page_test.rb:12]: Same here - can't really change this.

File: test/integration/platform_admin_deletes_job_test.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+class PlatformAdminDeletesJobTest < ActionDispatch::IntegrationTest
+  test "job and all related bids are deleted" do
+    bid = create(:bid)
+    bidder = bid.user
+    job = bid.job
+    job.bidding_open!
+    lister = job.lister
+    platform_admin = create(:platform_admin)
+
+    ApplicationController.any_instance.stubs(:current_user).returns(platform_admin)
```
Comment (ToniRib) [inline test/integration/platform_admin_deletes_job_test.rb:11]: Can't change this!

File: test/integration/platform_admin_view_category_show_test.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+require "test_helper"
+
+class PlatformAdminViewCategoryShowTest < ActionDispatch::IntegrationTest
+  test "platform admin can view category show page" do
+    job = create(:job)
+    platform_admin = create(:platform_admin)
+
+    ApplicationController.any_instance.stubs(:current_user).returns(platform_admin)
```
Comment (ToniRib) [inline test/integration/platform_admin_view_category_show_test.rb:8]: Can't change this!

File: app/presenters/job_presenter.rb [hunk 5]
```diff
@@ -38,14 +38,58 @@ def job_status_banner_or_bid_accordion
...
+    elsif !bidding_open? && bid.rejected?
+      view.content_tag :div, "Rejected", class: "status-rejected"
+    end
+  end
+
+  def delete_listing_button
+    if view.current_platform_admin?
+      view.content_tag(:div,
+                       view.link_to("Delete Job",
+                                    view.user_job_path(self.user, self),
```
Comment (ToniRib) [inline app/presenters/job_presenter.rb:78]: I actually need this so the path will be correct.

**Comment by ToniRib** [comment]
Merged in the latest master, this is ready for review!!

**Comment by ToniRib** [comment]
Pushed an extra commit to closes #116

File: test/integration/platform_admin_views_job_show_page_test.rb (hunk 2 above)
Comment (danjwinter) [inline test/integration/platform_admin_views_job_show_page_test.rb:12]: Hound just can't take a hint!

**Comment by danjwinter** [comment]
Everything looks good, loving the presenter objects!

**Pull request merged**